lib/gates_cirq.h

```diff
@@ -307,7 +315,11 @@ struct rz {
   static constexpr unsigned num_qubits = 1;
 
   static GateCirq<fp_type> Create(unsigned time, unsigned q0, fp_type phi) {
-    return ZPowGate<fp_type>::Create(time, q0, inv_pi * phi, -0.5);
+    fp_type c = std::cos(-0.5 * phi);
+    fp_type s = std::sin(-0.5 * phi);
+
+    return CreateGate<GateCirq<fp_type>, rz>(
```

Comment:
`GateKind` for this gate is still `krx` (should be `krz`)

Reply:
Done.